server/controllers/payroll/multiplePayroll/index.js

```diff
@@ -0,0 +1,1177 @@
+                    rubric.result = advantage.value * enterpriseExchangeRate;
+                    rubric.result = util.roundDecimal(rubric.result, DECIMAL_PRECISION);
+                  }
+                });
+              });
+
+              // Filtering non-taxable Rubrics
+              nonTaxables = rubricData.filter(item => item.is_social_care);
+
+              // Filtering taxable Rubrics
+              taxables = rubricData.filter(item =>
+                (item.is_tax !== 1 && item.is_social_care !== 1 && item.is_membership_fee !== 1));
+
+
+              // Filtering all taxes and contributions that is calculated from the taxable base
+              taxesContributions = rubricData.filter(item =>
+                item.is_tax || item.is_membership_fee || item.is_discount === 1);
+            }
+
+            // Calcul value for non-taxable and automatically calculated
+            if (nonTaxables.length) {
+              nonTaxables.forEach(nonTaxable => {
+                nonTaxable.result = nonTaxable.is_percent ?
+                  util.roundDecimal((basicSalary * nonTaxable.value) / 100, DECIMAL_PRECISION) :
+                  nonTaxable.result || nonTaxable.value;
+
+                sumNonTaxable += nonTaxable.result;
+
+                allRubrics.push([uid, nonTaxable.rubric_payroll_id, nonTaxable.result]);
+              });
+            }
+
+            if (taxables.length) {
+              taxables.forEach(taxable => {
+                taxable.result = taxable.is_percent ?
+                  util.roundDecimal((basicSalary * taxable.value) / 100, DECIMAL_PRECISION) :
+                  taxable.result || taxable.value;
+
+                sumTaxable += taxable.result;
+
+                allRubrics.push([uid, taxable.rubric_payroll_id, taxable.result]);
+              });
+            }
+
+            const baseTaxable = basicSalary + sumTaxable;
+
+            grossSalary = basicSalary + sumTaxable + sumNonTaxable;
+
+            if (taxesContributions.length) {
+              taxesContributions.forEach(taxContribution => {
+                taxContribution.result = taxContribution.is_percent ?
+                  util.roundDecimal((baseTaxable * taxContribution.value) / 100, DECIMAL_PRECISION) :
+                  taxContribution.result || taxContribution.value;
+
+                // Recovery of the value of the Membership Fee worker share
+                if (taxContribution.is_membership_fee && taxContribution.is_employee) {
+                  membershipFeeEmployee = taxContribution.result;
+                }
+              });
+            }
+
+            const baseIpr = ((baseTaxable - membershipFeeEmployee) * (iprExchangeRate / enterpriseExchangeRate));
+
+            // Annual cumulation of Base IPR
+            const annualCumulation = baseIpr * 12;
+
+            let ind = -1;
+            let iprValue = 0;
+            let scaleIndice;
+
+            if (iprScales.length) {
```

Comment:
This code looks identical to [line 796](https://github.com/IMA-WorldHealth/bhima-2.X/pull/2703/files#diff-7562bcd0f3ac8c515205012f560eaebdR796).   Is there a way we could make little functions that could be called twice instead of writing the code twice?